### PR TITLE
Separate cloud/non-cloud tests

### LINKED ISSUE #6588: Separate out functional tests that require cloud/external resources
### Area for Improvement (required)
Currently, all the functional tests run only with an approved "/ok-to-test" command. Only tests that deploy cloud resources or publish stuff really require an approval. This PR separates out the tests that do not need an approval to run as a part of the regular build workflow. The /ok-to-test command will now run only tests that deploy cloud resources

### Observed behavior (required)
A maintainer must issue an /ok-to-test command to run any functional test on the PR

### Desired behavior (required)
/ok-to-test command should be required to run only those tests that require cloud resources

### Proposed Fix (optional)
- Separate out the tests
- Modify the github workflows to run cloud/non-cloud tests

### rad Version (required)

0.26.3

### Operating system (required)
macOS

## Additional context

[AB#9945](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/9945)

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Currently, all the functional tests run only with an approved "/ok-to-test" command. Only tests that deploy cloud resources or publish stuff really require an approval. This PR separates out the tests that do not need an approval to run as a part of the regular build workflow. The /ok-to-test command will now run only tests that deploy cloud resources

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).

Fixes: #6588 

### Summary
🧪☁️🛠️

This pull request adds support for running functional tests that require cloud resources for the radius project. It modifies the GitHub workflows and the make targets to run different types of tests. It also moves the existing cloud-based tests to a new `test/functional-cloud` directory and adds new test files for the extender and Terraform recipe features.

> _To test radius with cloud resources_
> _We added some make targets and processes_
> _We use `rad` and `bicep`_
> _And `terraform` to equip_
> _Our workflow with functional successes_

### Walkthrough
*  Add a new step to the build workflow to run functional tests that do not require cloud resources ([link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1R115-R123))
*  Rename and modify the existing step to run functional tests that require cloud resources and append "-cloud" to the make command argument ([link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-c79f364a9293abaaa8595776b74674e24bec6287834e63ab8aa7aec6a42f0dbcL485-R485), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-c79f364a9293abaaa8595776b74674e24bec6287834e63ab8aa7aec6a42f0dbcL504-R504))
*  Add new make targets to run functional tests for shared, Datastores RP, and UCP components using the `test-functional-cloud` directory ([link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-607e82dd7ce2736355a7703fb1a4a832dddb294c3683c4b51a5118218e379712R69-R71), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-607e82dd7ce2736355a7703fb1a4a832dddb294c3683c4b51a5118218e379712R81-R83), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-607e82dd7ce2736355a7703fb1a4a832dddb294c3683c4b51a5118218e379712R90-R92))
*  Add links to the documentation for writing and running functional tests in the `test-functional-cloud/README.md` file ([link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-2b743c706c468ceba7f2e5216e5ba710b4ca05d7185fb81289c1502a5b2355d7R1-R3))
*  Add a new test file for testing the extender resource using a Bicep recipe that deploys an AWS S3 bucket (`test/functional-cloud/shared/resources/extender_test.go`, [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-bb6b1ed871e7e5ebca765d4e68671ae8fc5b32e463e0ced21695a9be107ce030R1-R105))
*  Add a new test file for testing the Terraform recipe functionality using the extender resource (`test/functional-cloud/shared/resources/recipe_terraform_test.go`, [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-3608e01a51c1e0dad64de358bf759b7ced3a52cdb6912f909488b546be37f1a8R1-R134))
*  Rename several test files and directories from `test/functional` to `test/functional-cloud` to reflect the cloud resource requirement ([link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-40d3dc97a9596407cb8b31753d6ac21890432bc774a4ed0dafdc9cca5ad11802), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-0b51a6fc41d0f9e59e35c11e1a5f2dcb8597075de36d264885e07c434e344679), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-0b37f569e289693587120696d3348772af575944e328dbbcb477df72262506ae), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-0d758a1298e2d687839ada31e80f435ccd7f7da82f0c13836b20558af143697b), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-f287609184b5be7f122670ad46055ff1501f21cf1ac1d595f21211ea1270ebfd), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-40927ffdde4082b11be4a6a93d5efb52417396e2512930d1d955000ad3ed6e4f), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-6a3091eab0d008959fce8f9a39c33f6765253859aea2ed54af5a5538fa2f783b), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-718c5afb2ef08f4903a373c110fdcf070955dc4beec70090a92c099be44a58f4), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-78d098ec612b44a169a739d4c0660150a33668a53ffd939ccbb1ed2198901e0e), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-276a5d47f470a09ed402699132bdd5a074bb550722755790564021dee3312373), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-a7e541ece632421aa4419114270f01f58321c85024a738b5b4f98f17a71c8089), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-2d54cb4987ad11c7e7f82944a10e7c9e54f23a2319a177f3d274e61dc0195735), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-5ee11009f0b52e32be460e0764058da18fa3878e9809e187635a217b8ed33b4b), [link](https://github.com/radius-project/radius/pull/6587/files?diff=unified&w=0#diff-d43aca1415129dd05ac608e1296762d1347c363cdad15113a8590f75327f0e73))

### CODE REVIEW

**Comment by vinayada1** [comment]
Looks like /ok-to-test still runs the old workflow and a PR merge would be needed for it to run only the cloud tests

File: test/functional-cloud/datastoresrp/resources/redis_test.go
Comment (sk593) [inline test/functional-cloud/datastoresrp/resources/redis_test.go:1]: Should the manual case be moved to the non-cloud tests?

**Review by kachawla — COMMENTED** [review]
Thanks for kicking this off!

As I mentioned in the design discussion, from this PR it looks like all the UCP tests only run on cloud resources, in which case we should add more tests for UCP that can run as a part of functional tests workflow. Can you please confirm this?

**Comment by vinayada1** [comment]
> Thanks for kicking this off!
> 
> As I mentioned in the design discussion, from this PR it looks like all the UCP tests only run on cloud resources, in which case we should add more tests for UCP that can run as a part of functional tests workflow. Can you please confirm this?

I checked it. There are other UCP tests for testing planes/RGs, etc. Since there is no change, they already exist in the same folder, they do not show up in the changes
Screenshot 2023-11-07 at 4 41 04 PM

**Pull request closed without merging**